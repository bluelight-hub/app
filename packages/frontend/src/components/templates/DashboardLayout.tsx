import React from 'react';
import { Outlet } from 'react-router';

/**
 * DashboardLayout-Template für Dashboards ohne Navigation
 * Wird für Fenster verwendet, die im Vollbildmodus oder als separates Fenster geöffnet werden
 * Nutzt die volle Fensterhöhe und macht den Inhalt bei Bedarf scrollbar
 */
const DashboardLayout: React.FC = () => {
    return (
        <div className="h-screen flex flex-col bg-white dark:bg-gray-900 text-black dark:text-white overflow-hidden">
            <main className="flex-1 overflow-auto py-4">
                <div className="px-4 h-full">
                    <Outlet />
                </div>
            </main>
        </div>
    );
};

export default DashboardLayout; 