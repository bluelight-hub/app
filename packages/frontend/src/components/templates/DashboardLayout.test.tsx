import { render, screen } from '@testing-library/react';
import { vi } from 'vitest';
import DashboardLayout from './DashboardLayout';

// Mock fÃ¼r Outlet
vi.mock('react-router', () => ({
    Outlet: () => <div data-testid="outlet-content">Dashboard Content</div>
}));

describe('DashboardLayout', () => {
    it('renders correctly with all components', () => {
        render(<DashboardLayout />);

        // Check if main content area is rendered
        expect(screen.getByTestId('outlet-content')).toBeInTheDocument();
    });

    it('has correct layout classes', () => {
        const { container } = render(<DashboardLayout />);

        // Check for correct container classes
        const layoutContainer = container.firstChild as HTMLElement;
        expect(layoutContainer).toHaveClass('min-h-screen');
        expect(layoutContainer).toHaveClass('bg-white');
        expect(layoutContainer).toHaveClass('dark:bg-gray-900');
    });

    // Snapshot test
    it('matches snapshot', () => {
        const { container } = render(<DashboardLayout />);
        expect(container).toMatchSnapshot();
    });
}); 