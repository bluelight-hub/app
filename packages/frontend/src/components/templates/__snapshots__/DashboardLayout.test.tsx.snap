// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DashboardLayout > matches snapshot 1`] = `
<div>
  <div
    class="min-h-screen flex flex-col bg-white dark:bg-gray-900 text-black dark:text-white overflow-hidden"
  >
    <main
      class="flex-1 overflow-auto py-4"
    >
      <div
        class="px-4 h-full"
      >
        <div
          data-testid="outlet-content"
        >
          Dashboard Content
        </div>
      </div>
    </main>
  </div>
</div>
`;
